src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -1299,6 +1299,42 @@ theorem saddOverflow_eq {w : Nat} (x y : BitVec w) :
     simp
     omega
 
+theorem usubOverflow_eq {w : Nat} (x y : BitVec w) :
+    usubOverflow x y = decide (x < y) := rfl
+
+theorem ssubOverflow_eq {w : Nat} (x y : BitVec w) :
+    ssubOverflow x y = ((!x.msb && y.msb && (x - y).msb) || (x.msb && !y.msb && !(x - y).msb)) := by
+  simp only [ssubOverflow]
+  rcases w with _|w
+  · revert x y; decide
+  · have := le_two_mul_toInt (x := x); have := two_mul_toInt_lt (x := x)
+    have := le_two_mul_toInt (x := y); have := two_mul_toInt_lt (x := y)
+    simp only [msb_eq_toInt, toInt_sub, Nat.add_one_sub_one, ge_iff_le]
+    have h₁ : (x.toInt - y.toInt < - 2 ^ w) ↔ (x.toInt < 0 ∧ 0 ≤ y.toInt ∧ 0 ≤ (x.toInt - y.toInt).bmod (2 ^ (w + 1))) := by 
+      constructor 
+      · intros h
+        simp only [show x.toInt < 0 by omega, show 0 ≤ y.toInt by omega, _root_.true_and]
+        rw [← Int.bmod_add_cancel]
+        rw_mod_cast [Int.bmod_eq_self_of_le (by omega) (by omega)]
+        omega 
+      · have := Int.bmod_neg_iff (x := x.toInt - y.toInt) (m := 2 ^ (w + 1))
+        push_cast at this
+        omega
+    have h₂ : (2 ^ w ≤ x.toInt - y.toInt) ↔ (0 ≤ x.toInt ∧ y.toInt < 0 ∧ (x.toInt - y.toInt).bmod (2 ^ (w + 1)) < 0) := by 
+      constructor 
+      · intros h 
+        simp only [show 0 ≤ x.toInt by omega, show y.toInt < 0 by omega, _root_.true_and]
+        rw [← Int.bmod_sub_cancel]
+        push_cast 
+        rw_mod_cast [Int.bmod_eq_self_of_le (by omega) (by omega)] 
+        omega 
+      · have := Int.bmod_neg_iff (x := x.toInt - y.toInt) (m := 2 ^ (w + 1))
+        push_cast at this 
+        omega
+    simp only [h₁, decide_and, h₂, ← decide_not, Int.not_lt]
+    simp only [bool_to_prop]
```

Comment:
```suggestion
    simp [h₁, decide_and, h₂, ← decide_not, Int.not_lt, bool_to_prop]
```

Reply:
this is not feasible, `bool_to_prop` needs to be on its own to avoid infinite recursion